avocado/core/parser.py

```diff
@@ -46,6 +51,32 @@ def _get_option_tuples(self, option_string):
         return []
 
 
+class StdoutAlreadyClaimedError(Exception):
+
+    """
+    Signals that other player already claimed exclusive rights on STDOUT
+    """
+
+
+class FileOrStdoutAction(argparse.Action):
```

Comment:
I have been meaning to change the stdout handling for a long time (see logging rework for details) and doing the fist check in `parser` is the best way (even better than I had in mind). The only thing I dislike on this PR is that you're using a global variable. I don't think it's needed, we can assume that one app is going to use just one parser instance, so I'd simply store the user in the `parser.stdout_claimed` (or even `stdout_claimed_by`). Doing so can allow other custom plugins to use this variable to do their checking (eg. when they don't support file and only support stdout replacement).

What do you think?